js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx

```diff
@@ -39,26 +43,41 @@ export const useAssetOwnerFilter = ({
 export function useAssetOwnersForAssets(
   assets: {
     definition?: {
-      owners: Array<
-        {__typename: 'TeamAssetOwner'; team: string} | {__typename: 'UserAssetOwner'; email: string}
-      >;
+      owners: AssetOwner[];
     } | null;
   }[],
-): string[] {
+): AssetOwner[] {
   return useMemo(
     () =>
       Array.from(
         new Set(
           assets
-            .flatMap(
-              (a) =>
-                a.definition?.owners.flatMap((o) =>
-                  o.__typename === 'TeamAssetOwner' ? o.team : o.email,
-                ),
-            )
-            .filter((o) => o),
+            .flatMap((a) => a.definition?.owners)
+            .filter((o) => o)
+            // Convert to JSON for deduping by Set
+            .map((owner) => JSON.stringify(owner)),
         ),
-      ) as string[],
+        // Convert back to AssetOwner
+      ).map((ownerJSON) => memoizedOwner(JSON.parse(ownerJSON))),
     [assets],
   );
 }
+
+const memoizedOwner = memoize(
+  (owner: AssetOwner) => {
+    return owner;
+  },
+  (owner) => JSON.stringify(owner),
+);
```

Comment:
Not really worth changing, but I think it'd be nice if this memoize function took the JSON string and returned the owner vs. taking an owner that has just been converted from a string and converting it back to a string to get a memo key 😅

Ala 

```
const memoizedOwnerFromJSON = memoize(
  (owner: string): AssetOwner  => {
    return JSON.parse(owner);
  }
);```